mantidproject/mantid
# Fix tabbing hidden columns

**Description of work.**
Fixes tabbing between hidden columns, this work includes:
- Moving hidden tabs to the end
- Reworking the SANS code base to not assume positional args
- Introducing a POD class to hold said data
- Updating the CSV parser to not expect an order of data
- Updating various unit tests

**To test:**
- Open the SANS Data GUI with LOQ data and attempt to break it
- For the sample shape issue; add some sample dimensions such as height / width, reload and check they are still there
- Check the various other tabs e.g. beam centre finder

Fixes #27028
Fixes #26912
Fixes #27169 
Fixes #27055 (this is hard to test, but the async race condition no longer exists)

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
